### PR TITLE
cpu: Expose SGX EPC resource

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

File: docs/usage/features.md [hunk 1]
```diff
@@ -55,7 +55,8 @@ option of nfd-worker.
 | **`cpu-pstate.scaling_governor`** | string | The value of the Intel pstate scaling_governor when in use, either 'powersave' or 'performance'.
 | **`cpu-cstate.enabled`**          | bool   | Set to 'true' if cstates are set in the intel_idle driver, otherwise set to 'false'. Unset if intel_idle cpuidle driver is not active.
 | **`cpu-rdt.<rdt-flag>`**          | true   | **DEPRECATED** [Intel RDT][intel-rdt] capability is supported. See [RDT flags](customization-guide.md#intel-rdt-flags) for details.
-| **`cpu-security.sgx.enabled`**    | true   | Set to 'true' if Intel SGX is enabled in BIOS (based a non-zero sum value of SGX EPC section sizes).
+| **`cpu-security.sgx.enabled`**    | true   | Set to 'true' if Intel SGX is enabled in BIOS (based on a non-zero sum value of SGX EPC section sizes).
+| **`cpu-security.sgx.epc`**        | int    | The total amount Intel SGX Encrypted Page Cache memory. It's only present if `sgx.enabled` is `true`.
```
Comment (marquiz) [inline docs/usage/features.md:59]: I'd say not expose this as a label by default. Just have it for NodeFeatureRules. WDYT?

Just realized this aspect of the tdx.total_keys and submitted #1130 to alter that.

File: docs/usage/customization-guide.md [hunk 2]
```diff
@@ -675,6 +675,7 @@ The following features are available for matching:
 |                  |              | **`RDTL3CA_NUM_CLOSID`** | int  | The number or available CLOSID (Class of service ID) for Intel L3 Cache Allocation Technology
 | **`cpu.security`** | attribute  |          |            | Features related to security and trusted execution environments
 |                  |              | **`sgx.enabled`** | bool | `true` if Intel SGX (Software Guard Extensions) has been enabled, otherwise does not exist
+|                  |              | **`sgx.epc`** | int | The total amount Intel SGX Encrypted Page Cache memory. It's only present if `sgx.enabled` is `true`.
```
Comment (marquiz) [inline docs/usage/customization-guide.md:678]: What is the unit of this? Bytes?

Comment (mythi) [reply docs/usage/customization-guide.md:678]: yes, I can add a note about that.

File: docs/usage/features.md (hunk 1 above)
Comment (mythi) [reply docs/usage/features.md:59]: I thought about it too but did not get to addressing that yet. Previously, that "filtering" was done by `-resource-labels` and sources were free to create whatever labels they wanted. Thanks for #1130

Comment (marquiz) [reply docs/usage/features.md:59]: K, so maybe wait for #1130 to be merged and then rebase.

Comment (mythi) [reply docs/usage/features.md]: sounds good. I already dropped this `features.md` addition.

**Comment by marquiz** [comment]
@mythi you can now rebase
/ok-to-test

File: docs/usage/features.md (hunk 1 above)
Comment (mythi) [reply docs/usage/features.md]: rebased

**Comment by mythi** [comment]
Tested with the following setup and it works as expected:

```
apiVersion: nfd.k8s-sigs.io/v1alpha1
kind: NodeFeatureRule
metadata:
  name: sgx-epc-rule
spec:
  rules:
  - name: "SGX EPC"
    extendedResources:
      sgx.intel.com/epc2: "@cpu.security.sgx.epc"
    matchFeatures:
      - feature: cpu.security
        matchExpressions:
          sgx.enabled: {op: IsTrue}
```

**Review by marquiz — APPROVED** [review]
Thx @mythi! Just in time for v0.13 😎

**Pull request merged**